mainwindow.cpp

```diff
@@ -169,5 +175,47 @@ void MainWindow::mapping_card_value()
     else {
         cardArr &= ~(1UL << 15);
     }
-//    if (cardArr.count)
+
+    //handle move count
+    oldCardArr ^= cardArr;
+    for(; oldCardArr>0;)
+    {
+        if(oldCardArr%2)
+        {
+            numberOfMove++;
+            numberOfOpenCard++;
+        }
+        oldCardArr = oldCardArr >> 1;
+        if(numberOfOpenCard >= 2)
+        {
+            numberOfOpenCard = 0;
+            compare_card(cardArr);
+        }
+    }
+    ui->moveNumber->display(numberOfMove);
+};
+
+void MainWindow::change_card_availability(bool state)
+{
+    ui->cardCheckBox->setCheckable(state);
+    ui->cardCheckBox_2->setCheckable(state);
+    ui->cardCheckBox_3->setCheckable(state);
+    ui->cardCheckBox_4->setCheckable(state);
+    ui->cardCheckBox_5->setCheckable(state);
+    ui->cardCheckBox_6->setCheckable(state);
+    ui->cardCheckBox_7->setCheckable(state);
+    ui->cardCheckBox_8->setCheckable(state);
+    ui->cardCheckBox_9->setCheckable(state);
+    ui->cardCheckBox_10->setCheckable(state);
+    ui->cardCheckBox_11->setCheckable(state);
+    ui->cardCheckBox_12->setCheckable(state);
+    ui->cardCheckBox_13->setCheckable(state);
+    ui->cardCheckBox_14->setCheckable(state);
+    ui->cardCheckBox_15->setCheckable(state);
+    ui->cardCheckBox_16->setCheckable(state);
+};
+
+void MainWindow::compare_card(uint arr)
```

Comment:
I think we should split this one into 2 functions:

-` bool is_matched_cards(card1, card2)` to check if the two cards are a matched or not? (I'm not very sure about the type for card1 and card2 though)
- `void disable_card(card)` to disable the matched cards